rerun_cpp/CMakeLists.txt

```diff
@@ -168,18 +174,20 @@ if(RERUN_DOWNLOAD_AND_BUILD_ARROW)
         GIT_PROGRESS true
         CMAKE_ARGS
         --preset ninja-debug-minimal
-        -DARROW_IPC=ON
+        -DARROW_BOOST_USE_SHARED=OFF
         -DARROW_BUILD_SHARED=${ARROW_BUILD_SHARED}
         -DARROW_BUILD_STATIC=${ARROW_BUILD_STATIC}
-        -DCMAKE_INSTALL_PREFIX=${ARROW_DOWNLOAD_PATH}
+        -DARROW_CXXFLAGS=${DARROW_CXXFLAGS}
+        -DARROW_IPC=ON
+        -DARROW_JEMALLOC=OFF
+        -DARROW_MIMALLOC=ON
```

Comment:
is mimalloc automatically built from source, or will this cause new linking issues?

Reply:
Yeah, it does it's own fetch + build, the same way we do with arrow:
```
[6/145] Performing download step (download, verify and extract) for 'mimalloc_ep'
-- Downloading...
   dst='/home/jleibs/cpp-example-vrs/build/arrow/src/arrow_cpp-build/mimalloc_ep-prefix/src/v2.0.6.tar.gz'
   timeout='none'
   inactivity timeout='none'
-- Using src='https://github.com/microsoft/mimalloc/archive/v2.0.6.tar.gz'
-- verifying file...
       file='/home/jleibs/cpp-example-vrs/build/arrow/src/arrow_cpp-build/mimalloc_ep-prefix/src/v2.0.6.tar.gz'
-- Downloading... done
-- extracting...
     src='/home/jleibs/cpp-example-vrs/build/arrow/src/arrow_cpp-build/mimalloc_ep-prefix/src/v2.0.6.tar.gz'
     dst='/home/jleibs/cpp-example-vrs/build/arrow/src/arrow_cpp-build/mimalloc_ep-prefix/src/mimalloc_ep'
```